coax/td_learning/_doubleqlearning.py

```diff
@@ -143,10 +143,10 @@ def q_targ_func(self, target_params, target_state, rng, transition_batch):
             # get greedy action as the mode of pi_targ
             params, state = target_params['pi_targ'], target_state['pi_targ']
             S_next = self.pi_targ.observation_preprocessor(next(rngs), transition_batch.S_next)
+            dist_params, _ = self.pi_targ.function(params, state, next(rngs), S_next, False)
             if self.greedy_pi_targ:
-                A_next = self.pi_targ.mode_func(params, state, next(rngs), S_next)
+                A_next = self.pi_targ.proba_dist.mode(dist_params)
             else:
-                dist_params, _ = self.pi_targ.function(params, state, next(rngs), S_next, False)
                 A_next = self.pi_targ.proba_dist.sample(dist_params, next(rngs))
```

Comment:
Yes, you are correct! Most implementations (such as [spinup](https://github.com/openai/spinningup/blob/master/spinup/algos/pytorch/sac/core.py#L51)) do not distinguish between taking the mode or a sample from the current policy. So to stay in line with the framework of staying close to the theory, we can either introduce a `DoubleSARSA` updater or always perform the greedy action selection in the `*QLearning` updater. For this PR I would prefer the latter, because it does introduce fewer changes.